presto-main/src/main/java/com/facebook/presto/cost/HistoryBasedStatisticsCacheManager.java

```diff
@@ -40,26 +43,40 @@ public class HistoryBasedStatisticsCacheManager
     // Cache hashes of plan node.
     private final Map<QueryId, Map<CachingPlanCanonicalInfoProvider.CacheKey, PlanNodeCanonicalInfo>> canonicalInfoCache = new ConcurrentHashMap<>();
 
+    // Whether the history is loaded successfully
+    private final Set<QueryId> invalidQueryId = ConcurrentHashMap.newKeySet();
```

Comment:
why is this is a set? do we expect to have more than 1 item there? or is it so we can quickly check for containment without worrying if it is set or not

Reply:
The query ID will be removed after the query completes. Yes, there can be more than 1 item, when multiple queries are running concurrently.

The life span of the query ID will be the same as in other Maps